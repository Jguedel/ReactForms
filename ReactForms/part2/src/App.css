.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.item {
  margin: 5px auto 5px auto;
  max-width: 500px;
  outline: solid 2px black;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  text-align: center;
}
.delBtn {
  background-color: red;
  width: 25px;
  height: 25px;
  align-self: center;
  margin: auto 10px auto 10px;
  text-align: center;
  border-radius: 8px;
}
.delBtn:hover {
  background-color: rgb(112, 0, 0);
  cursor: pointer;
}
.taskText {
  text-decoration: underline;
  font-weight: bold;
  font-size: 20px;
}

.todoForm {
  display: flex;
  justify-content: center;
}
.subBtn {
  background-color: rgb(165, 164, 164);
  margin-left: 10px;
  border-radius: 8px;
}
.subBtn:hover {
  background-color: rgb(46, 46, 46);
  cursor: pointer;
}
