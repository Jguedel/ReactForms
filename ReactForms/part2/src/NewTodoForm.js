import React, { useState } from "react";
import { v4 as uuid } from "uuid";
const NewTodoForm = (props) => {
  const INITIAL = {
    task: "",
  };
  const [formData, setformData] = useState(INITIAL);
  const handleChange = (e) => {
    const { name, value } = e.target;
    setformData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    props.addItem(formData.task, uuid());
    setformData(INITIAL);
  };
  return (
    <div>
      <form className="todoForm" onSubmit={handleSubmit}>
        <div>
          <label htmlFor="task">Task: </label>
          <input
            className="task"
            name="task"
            type="text"
            placeholder="Enter your task"
            value={formData.task}
            onChange={handleChange}
            required
          />
        </div>
        <button className="subBtn">Add!</button>
      </form>
    </div>
  );
};
export default NewTodoForm;
