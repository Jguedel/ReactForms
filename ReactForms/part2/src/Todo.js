const Todo = (props) => {
  const handleClick = (e) => {
    e.preventDefault();
    props.removeItem(props.task, props.id);
  };
  return (
    <div className="item">
      <p className="taskText">{props.task}</p>
      <button className="delBtn" onClick={handleClick}>
        X
      </button>
    </div>
  );
};
export default Todo;
