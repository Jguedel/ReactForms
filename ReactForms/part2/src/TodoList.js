import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  const lists = [
    {
      task: "you can delete the task by pushing the X ->",
      key: 156,
    },
  ];
  const [items, setItems] = useState(lists);
  const addItem = (task, key) => {
    setItems((items) => [...items, { task, key }]);
  };
  const removeItem = (task, key) => {
    setItems(items.filter((item) => key != item.key));
  };

  return (
    <div className="boxList">
      <h1>Todo List</h1>
      <NewTodoForm addItem={addItem} key={1} />
      {items.map((item) => {
        return (
          <Todo
            key={item.key}
            id={item.key}
            task={item.task}
            removeItem={removeItem}
          />
        );
      })}
    </div>
  );
};
export default TodoList;
