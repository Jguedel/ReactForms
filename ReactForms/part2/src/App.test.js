import { render, fireEvent } from "@testing-library/react";
import App from "./App";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import TodoList from "./TodoList";

test("App renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("Todo renders", () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

test("TodoList renders", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("NewTodoForm renders", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

//Adding/Removing item
test("Adding and removing a task", () => {
  //starting with nothing
  const { container } = render(<TodoList />);
  expect(container.querySelectorAll(".item")).length = 1;
  const text = container.querySelector(".task");
  const btn = container.querySelector(".subBtn");

  //add one item
  test.value = "test";
  fireEvent.click(btn);
  expect(container.querySelectorAll(".item")).length = 2;
  expect(text).value = "";
  expect(container.querySelectorAll(".item")[1]).value = "test";

  //add one item
  const delbtn = container.querySelectorAll(".item")[1];
  fireEvent.click(delbtn);
  expect(container.querySelectorAll(".item")).length = 1;
  expect(text).value = "";
});
