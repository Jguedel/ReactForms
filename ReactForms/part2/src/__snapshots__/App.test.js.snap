// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="boxList"
    >
      <h1>
        Todo List
      </h1>
      <div
        class="todoForm"
      >
        <form>
          <div>
            <label
              for="task"
            >
              Task: 
            </label>
            <input
              class="task"
              name="task"
              placeholder="Enter your task"
              type="text"
              value=""
            />
          </div>
          <button
            class="subBtn"
          >
            Add!
          </button>
        </form>
      </div>
      <div
        class="item"
      >
        <p
          class="task"
        >
          you can delete the task by pushing the X
        </p>
        <button
          class="delBtn"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewTodoForm renders 1`] = `
<DocumentFragment>
  <div
    class="todoForm"
  >
    <form>
      <div>
        <label
          for="task"
        >
          Task: 
        </label>
        <input
          class="task"
          name="task"
          placeholder="Enter your task"
          type="text"
          value=""
        />
      </div>
      <button
        class="subBtn"
      >
        Add!
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Todo renders 1`] = `
<DocumentFragment>
  <div
    class="item"
  >
    <p
      class="task"
    />
    <button
      class="delBtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`TodoList renders 1`] = `
<DocumentFragment>
  <div
    class="boxList"
  >
    <h1>
      Todo List
    </h1>
    <div
      class="todoForm"
    >
      <form>
        <div>
          <label
            for="task"
          >
            Task: 
          </label>
          <input
            class="task"
            name="task"
            placeholder="Enter your task"
            type="text"
            value=""
          />
        </div>
        <button
          class="subBtn"
        >
          Add!
        </button>
      </form>
    </div>
    <div
      class="item"
    >
      <p
        class="task"
      >
        you can delete the task by pushing the X
      </p>
      <button
        class="delBtn"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
